### PR TITLE
Fixes #394 Navbar uses angular topmenu API 

### LINKED ISSUE #394: generic topmenu
all the html files in loklak have a top-menu navigation with looks like (here: architecture.html):
```
<div id="navbar" class="navbar-collapse collapse">
          <ul class="nav navbar-nav navbar-right">
            <li><a href="index.html">Home</a></li>
            <li><a href="about.html">About</a></li>
            <li><a href="showcase.html">Showcase</a></li>
            <li class="active"><a href="architecture.html">Architecture</a></li>
            <li><a href="download.html">Download</a></li>
            <li><a href="tutorials.html">Tutorials</a></li>
            <li><a href="api.html">API</a></li>
            <li><a href="dump.html">Dumps</a></li>
            <li><a href="apps/">Apps</a></li>
          </ul>
```
If a new menu item is created or deleted, all menu entries in all html files must be changed. This shall be changed using angular and the new topmenu servlet introduced in https://github.com/loklak/loklak_server/commit/5177b4df4b42dfdc7e545afbed853decc7b9365b

currently /cms/topmenu.json creates:
```
{"items": [
  {"Home": "index.html"},
  {"About": "about.html"},
  {"Showcase": "showcase.html"},
  {"Architecture": "architecture.html"},
  {"Download": "download.html"},
  {"Tutorials": "tutorials.html"},
  {"API": "api.html"},
  {"Dumps": "dump.html"},
  {"Apps": "apps/"}
]}
```

- change all html files with top-menue (including all apps!)
- read /cms/topmenu.json
- all html files must identify the own html pages name and write an ```class="active"```into the ```li``` options
- if this requires additional js code, write it into https://github.com/loklak/loklak_server/blob/master/html/js/loklak.js

### PR DESCRIPTION
**Pull request by daminisatya**

### CODE REVIEW

**Comment by daminisatya** [comment]
Corrected the landing page!

File: html/js/loklak.js [hunk 1]
```diff
@@ -21,3 +21,68 @@ app.controller("search", function($scope, $http) {
     }
   }
 });
+
+app.controller("topmenu", function($scope, $http) {
```
Comment (sudheesh001) [inline html/js/loklak.js]: Codacy Issue found: [Strings must use doublequote.](https://www.codacy.com/app/sudheesh1995/loklak_server/file/2616038392/issues/source?bid=3288751&fileBranchId=3312976#l26)

File: html/js/loklak.js [hunk 2]
```diff
@@ -21,3 +21,68 @@ app.controller("search", function($scope, $http) {
...
+    when('/about', {
+        controller: 'widgetsController',
+        activetab: 'About'
+    }).
+    when('/showcase', {
+        controller: 'widgetsController',
+        activetab: 'Showcase'
+    }).
+    when('/architecture', {
+        controller: 'widgetsController',
```
Comment (sudheesh001) [inline html/js/loklak.js]: Codacy Issue found: [Strings must use doublequote.](https://www.codacy.com/app/sudheesh1995/loklak_server/file/2616038392/issues/source?bid=3288751&fileBranchId=3312976#l59)

File: html/js/loklak.js [hunk 3]
```diff
@@ -21,3 +21,68 @@ app.controller("search", function($scope, $http) {
...
+    }
+    return items.slice().reverse();
+  };
+});
+
+angular.module('loklak')
+.config(function ($routeProvider) {
+    $routeProvider.
+    when('/index', {
+        controller: 'widgetsController',
```
Comment (sudheesh001) [inline html/js/loklak.js]: Codacy Issue found: [Strings must use doublequote.](https://www.codacy.com/app/sudheesh1995/loklak_server/file/2616038392/issues/source?bid=3288751&fileBranchId=3312976#l47)

File: html/js/loklak.js [hunk 4]
```diff
@@ -21,3 +21,68 @@ app.controller("search", function($scope, $http) {
...
+        activetab: 'Home'
+    }).
+    when('/about', {
+        controller: 'widgetsController',
```
Comment (sudheesh001) [inline html/js/loklak.js]: Codacy Issue found: [Strings must use doublequote.](https://www.codacy.com/app/sudheesh1995/loklak_server/file/2616038392/issues/source?bid=3288751&fileBranchId=3312976#l51)

File: html/js/loklak.js [hunk 5]
```diff
@@ -21,3 +21,68 @@ app.controller("search", function($scope, $http) {
...
+        activetab: 'About'
+    }).
+    when('/showcase', {
+        controller: 'widgetsController',
```
Comment (sudheesh001) [inline html/js/loklak.js]: Codacy Issue found: [Strings must use doublequote.](https://www.codacy.com/app/sudheesh1995/loklak_server/file/2616038392/issues/source?bid=3288751&fileBranchId=3312976#l55)

File: html/js/loklak.js [hunk 6]
```diff
@@ -21,3 +21,68 @@ app.controller("search", function($scope, $http) {
...
+        activetab: 'Showcase'
+    }).
```
Comment (sudheesh001) [inline html/js/loklak.js]: Codacy Issue found: [Strings must use doublequote.](https://www.codacy.com/app/sudheesh1995/loklak_server/file/2616038392/issues/source?bid=3288751&fileBranchId=3312976#l57)

File: html/js/loklak.js [hunk 7]
```diff
@@ -21,3 +21,68 @@ app.controller("search", function($scope, $http) {
...
+    when('/tutorials', {
+        controller: 'widgetsController',
+        activetab: 'Tutorials'
+
+    }).
+    when('/api', {
+        controller: 'widgetsController',
+        activetab: 'API'
+    }).
```
Comment (sudheesh001) [inline html/js/loklak.js]: Codacy Issue found: [Strings must use doublequote.](https://www.codacy.com/app/sudheesh1995/loklak_server/file/2616038392/issues/source?bid=3288751&fileBranchId=3312976#l75)

File: html/js/loklak.js [hunk 8]
```diff
@@ -21,3 +21,68 @@ app.controller("search", function($scope, $http) {
...
+    if (!items || !items.length) {
+      return;
+    }
+    return items.slice().reverse();
+  };
+});
+
+angular.module('loklak')
+.config(function ($routeProvider) {
+    $routeProvider.
```
Comment (sudheesh001) [inline html/js/loklak.js]: Codacy Issue found: [Strings must use doublequote.](https://www.codacy.com/app/sudheesh1995/loklak_server/file/2616038392/issues/source?bid=3288751&fileBranchId=3312976#l45)

File: html/js/loklak.js [hunk 9]
```diff
@@ -21,3 +21,68 @@ app.controller("search", function($scope, $http) {
...
+        controller: 'widgetsController',
+        activetab: 'Tutorials'
+
+    }).
+    when('/api', {
+        controller: 'widgetsController',
+        activetab: 'API'
+    }).
+    when('/dump', {
+        controller: 'widgetsController',
```
Comment (sudheesh001) [inline html/js/loklak.js]: Codacy Issue found: [Strings must use doublequote.](https://www.codacy.com/app/sudheesh1995/loklak_server/file/2616038392/issues/source?bid=3288751&fileBranchId=3312976#l77)

File: html/js/loklak.js [hunk 10]
```diff
@@ -21,3 +21,68 @@ app.controller("search", function($scope, $http) {
...
+      return;
+    }
+    return items.slice().reverse();
+  };
+});
+
+angular.module('loklak')
+.config(function ($routeProvider) {
+    $routeProvider.
+    when('/index', {
```
Comment (sudheesh001) [inline html/js/loklak.js]: Codacy Issue found: [Strings must use doublequote.](https://www.codacy.com/app/sudheesh1995/loklak_server/file/2616038392/issues/source?bid=3288751&fileBranchId=3312976#l46)

File: html/js/loklak.js (hunk 2 above)
Comment (sudheesh001) [inline html/js/loklak.js]: Codacy Issue found: [Strings must use doublequote.](https://www.codacy.com/app/sudheesh1995/loklak_server/file/2616056478/issues/source?bid=3288751&fileBranchId=3312976#l59)

File: html/js/loklak.js (hunk 1 above)
Comment (sudheesh001) [inline html/js/loklak.js]: Codacy Issue found: [Strings must use doublequote.](https://www.codacy.com/app/sudheesh1995/loklak_server/file/2616056478/issues/source?bid=3288751&fileBranchId=3312976#l26)

File: html/js/loklak.js (hunk 3 above)
Comment (sudheesh001) [inline html/js/loklak.js]: Codacy Issue found: [Strings must use doublequote.](https://www.codacy.com/app/sudheesh1995/loklak_server/file/2616056478/issues/source?bid=3288751&fileBranchId=3312976#l47)

File: html/js/loklak.js (hunk 4 above)
Comment (sudheesh001) [inline html/js/loklak.js]: Codacy Issue found: [Strings must use doublequote.](https://www.codacy.com/app/sudheesh1995/loklak_server/file/2616056478/issues/source?bid=3288751&fileBranchId=3312976#l51)

File: html/js/loklak.js (hunk 6 above)
Comment (sudheesh001) [inline html/js/loklak.js]: Codacy Issue found: [Strings must use doublequote.](https://www.codacy.com/app/sudheesh1995/loklak_server/file/2616056478/issues/source?bid=3288751&fileBranchId=3312976#l57)

File: html/js/loklak.js (hunk 5 above)
Comment (sudheesh001) [inline html/js/loklak.js]: Codacy Issue found: [Strings must use doublequote.](https://www.codacy.com/app/sudheesh1995/loklak_server/file/2616056478/issues/source?bid=3288751&fileBranchId=3312976#l55)

File: html/js/loklak.js (hunk 7 above)
Comment (sudheesh001) [inline html/js/loklak.js]: Codacy Issue found: [Strings must use doublequote.](https://www.codacy.com/app/sudheesh1995/loklak_server/file/2616056478/issues/source?bid=3288751&fileBranchId=3312976#l75)

File: html/js/loklak.js (hunk 8 above)
Comment (sudheesh001) [inline html/js/loklak.js]: Codacy Issue found: [Strings must use doublequote.](https://www.codacy.com/app/sudheesh1995/loklak_server/file/2616056478/issues/source?bid=3288751&fileBranchId=3312976#l45)

File: html/js/loklak.js (hunk 9 above)
Comment (sudheesh001) [inline html/js/loklak.js]: Codacy Issue found: [Strings must use doublequote.](https://www.codacy.com/app/sudheesh1995/loklak_server/file/2616056478/issues/source?bid=3288751&fileBranchId=3312976#l77)

File: html/js/loklak.js (hunk 10 above)
Comment (sudheesh001) [inline html/js/loklak.js]: Codacy Issue found: [Strings must use doublequote.](https://www.codacy.com/app/sudheesh1995/loklak_server/file/2616056478/issues/source?bid=3288751&fileBranchId=3312976#l46)

**Comment by Orbiter** [comment]
The angular code looks good to me, what does "Do not merge" mean?

**Comment by sudheesh001** [comment]
@Orbiter I think the do not merge is because it's only done for the `index.html` page and not for the other pages, Once that's done this is good to merge. 😄

File: html/js/loklak.js [hunk 11]
```diff
@@ -21,3 +21,68 @@ app.controller("search", function($scope, $http) {
...
+app.filter("reverse", function() {
+  return function(items) {
+    if (!items || !items.length) {
+      return;
+    }
+    return items.slice().reverse();
+  };
+});
+
+angular.module("loklak")
```
Comment (sudheesh001) [inline html/js/loklak.js]: Codacy Issue found: ["angular" is not defined.](https://www.codacy.com/app/sudheesh1995/loklak_server/file/2623295948/issues/source?bid=3288751&fileBranchId=3312976#l42)

File: html/js/loklak.js [hunk 12]
```diff
@@ -21,3 +21,68 @@ app.controller("search", function($scope, $http) {
     }
   }
 });
+
+app.controller("topmenu", function($scope, $http) {
+  $http.get("/cms/topmenu.json").
+    success(function(data, status, headers, config){
+        $scope.results = data.items;
+    });
+    console.log($scope);
```
Comment (sudheesh001) [inline html/js/loklak.js]: Codacy Issue found: [Unexpected console statement.](https://www.codacy.com/app/sudheesh1995/loklak_server/file/2623295948/issues/source?bid=3288751&fileBranchId=3312976#l30)

File: html/js/loklak.js [hunk 13]
```diff
@@ -21,3 +21,67 @@ app.controller("search", function($scope, $http) {
...
+app.filter("reverse", function() {
+  return function(items) {
+    if (!items || !items.length) {
+      return;
+    }
+    return items.slice().reverse();
+  };
+});
+
+angular.module("loklak")
```
Comment (sudheesh001) [inline html/js/loklak.js]: Codacy Issue found: ["angular" is not defined.](https://www.codacy.com/app/sudheesh1995/loklak_server/file/2623356534/issues/source?bid=3288751&fileBranchId=3312976#l41)

**Pull request merged**